BusinessLogic/src/main/java/BusinessLogicLayer/SynchronizedTermsMap.java

```diff
@@ -5,33 +5,44 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
-import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReentrantLock;
 import java.util.stream.Collectors;
 
 /**
  * Created by HassanMahmud on 05/03/2016.
  */
 public class SynchronizedTermsMap {
-    public final static int TERM_NOT_IN_MAP_ERROR = -1;
-    public Lock lock = new ReentrantLock();
-    private Map<String, Integer> syncTermsMap;
+    private IncrementalIDGenerator incrementalIDGenerator;
+    private Map<String, String> syncTermsMap;
 
-    public SynchronizedTermsMap(List<Term> terms) {
-        Map<String, Integer> termsMap = terms.stream().collect(Collectors.toMap(Term::getTerm_Value, Term::getTerm_ID));
+    public SynchronizedTermsMap(List<Term> terms, IncrementalIDGenerator incrementalIDGenerator) {
+        this.incrementalIDGenerator = incrementalIDGenerator;
+        Map<String, String> termsMap = terms.stream().collect(
+                Collectors.toMap(Term::getTerm_Value, Term::getTerm_ID));
         syncTermsMap = Collections.synchronizedMap(termsMap);
     }
 
-    public synchronized int getTermID(String term_Value) {
-        boolean containsTermValue = syncTermsMap.containsKey(term_Value);
-        if (containsTermValue) {
-            Integer termId = syncTermsMap.get(term_Value);
-            return termId;
-        } else
-            return TERM_NOT_IN_MAP_ERROR;
+    public synchronized int getMapSize(){
+        return syncTermsMap.size();
     }
+//
+//    public synchronized String checkDuplicateTerms(String term_value, String generatedTermId) {
+//        boolean containsTermValue = syncTermsMap.containsKey(term_value);
+//        if (containsTermValue) {
+//            return syncTermsMap.get(term_value);
+//        } else {
+//            syncTermsMap.put(term_value, generatedTermId);
+//            return generatedTermId;
+//        }
+//    }
 
-    public synchronized void putTerm(Term term) {
-        syncTermsMap.put(term.getTerm_Value(), term.getTerm_ID());
+    public synchronized String getIdOrGenerateNewId(String term_value) {
+        boolean containsTermValue = syncTermsMap.containsKey(term_value);
+        if (containsTermValue) {
+            return syncTermsMap.get(term_value);
```

Comment:
Here, even though you find the term already in the map, you return it's _existing_ ID.